Documentation/design-docs/host-startup-hook.md

```diff
@@ -93,4 +94,119 @@ namespace SharedHostPolicy
         }
     }
 }
-```
\ No newline at end of file
+```
+
+## Error handling details
+
+Problems with the startup hook should be fairly straightforward to
+diagnose. All of these exceptionts will contain the startup hook path
+(`System.StartupHookProvider.ProcessStartupHooks`) on the stack
+trace. They fall into the following categories:
+
+- Errors detected eagerly, with exceptions thrown before the execution
+  of any startup hook.
+
+-- Invalid syntax. This will throw an `ArgumentException` with "The
+  syntax of the startup hook variable was invalid.".
+
+-- Partially qualified path in the startup hook. This will throw an
+   `ArgumentException` with "Absolute path information is required.".
+
+- Exceptions thrown during the call to a given startup hook. Previous
+  hooks may have run successfully.
+
+-- Missing startup hook assembly. This will throw a
+   `FileNotFoundException` with "Could not load file or assembly
+   '<assemblyPath>'".
+
+-- Invalid startup hook assembly. This will throw a
+   `BadImageFormatException` with the startup hook path on the stack
+   trace.
+
+-- Missing startup hook type. This will throw a `TypeLoadException`
+   with "Could not load type '<specifiedType>' from assembly
+   '<assemblyName>'".
+
+-- Missing `Initialize` method in startup hook. This will throw a
+   `MissingMethodException` with "Method '<specifiedType>.Initialize'
+   not found.".
+
+-- Invalid `Initialize` method (with an incorrect signature - one that
+   takes parameters, has a non-void return type, is not public, or is
+   not static). This will throw an `ArgumentException` with "The
+   signature of the startup hook '<specifiedType>.Initialize was
+   invalid. It must be 'public static void Initialize()'.
+
+## Guidance and caveats
+
+This hook is meant as a low-level, powerful way to inject code into
+the process at runtime, for use by tool developers who truly have a
+need for this kind of power. It should only be used in situations
+where modifying application code is not an option and there is not an
+existing structured dependency injection framework in place. An
+example of such a use case is a hook that injects logging, telemetry,
+or profiling into an existing deployed application at run tim.e
+
+It is prone to ordering issues when multiple hooks are used, and does
+nothing to attempt to make dependencies of hooks easy to
+manage. Multiple hooks, if used, should be fairly independent of each
```

Comment:
"Multiple hooks should be independent of each other" <- This should be a very strong statement with no ambiguity.